main.go

```diff
@@ -115,6 +115,13 @@ func onFileOpenButtonClickedHandler() {
 }
 
 func onAnalyzeButtonClickedHandler() {
+	// Prevent re-entry if already processing
+	if criticWindow.IsAnalyzeButtonActive {
+		dialog.ShowInformation("Analyze in Progress", "Analysis is already in progress. Please wait until the current process is complete.", *criticWindow.Window)
```

Comment:
Should we just disable the button?

Reply:
I tried this- unfortunately, I couldn't find a way to get the instance of the button from the `criticWindow.Toolbar` with the current implementation

Reply:
Hmm, that is a design mistake on my part then. 

This is fine for now given that constraint and I'll plan for a refactor of the critic window struct.